code/datums/flock/flockprotection.dm

```diff
@@ -1,76 +1,82 @@
+
+/// Mark a specific flock as interested in this
+/datum/component/flock_interest
+	/// The flock who is intently interested in this thing.
+	var/datum/flock/flock
+
+/datum/component/flock_interest/Initialize(datum/flock/flock)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.flock = flock
+
+/datum/component/flock_interest/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_FLOCK_ATTACK, .proc/handle_flock_attack)
+
+/// If flockdrone is in our flock, deny the attack, otherwise scream and cry
+/datum/component/flock_interest/proc/handle_flock_attack(source, mob/user)
+	if(!istype(user))
+		return FALSE
+
+	var/mob/living/critter/flock/F = user
+	if (istype(F) && F.flock == src.flock)
+		boutput(user, "<span class='alert'>The grip tool refuses to harm this, jamming briefly.</span>")
+		return TRUE
+
+	var/mob/living/critter/flock/drone/snitch
+	for (var/mob/living/critter/flock/drone/FD in view(7, source))
+		if (FD != user && !isdead(FD) && FD.is_npc && FD.flock == src.flock)
+			snitch = FD
+			break
+
+	if(!snitch)
+		return
+
+	if (!snitch.flock.isEnemy(user))
+		flock_speak(snitch, "Damage sighted on [source], [pick_string("flockmind.txt", "flockdrone_enemy")] [user]", snitch.flock)
+		snitch.flock.updateEnemy(user)
```

Comment:
```suggestion
	snitch.flock.updateEnemy(user)
```
Didn't realize when making PR that updateEnemy() does more than just add enemy icon